### ISSUE TITLE
Pagination metadata in search response

### ISSUE THREAD

**Issue by kbgg** [radiantearth/stac-spec#469]
It'd be useful to have information about which page the result is returned for, how many pages there are, how many results are returned, and the results returned per page.

For the QGIS plugin I'm showing the user a progress bar as search requests are made and without information about how many pages I'll be searching there's no way to show the user anything other than an indeterminate progress bar.

Sat-api has this implemented, albeit without the total pages, but I think it's a good starting point.

I'm proposing a simple addition to the response in searches which would look like this:
```
"meta" : {
    "page": 1,
    "total_pages": 102785,
    "limit": 10,
    "found": 1027850,
    "returned": 10
}
```

**Comment by m-mohr** [comment]
`total_pages` and `found` may not be available in some environments. What is `returned` useful for?

**Comment by p3dr0** [comment]
for pagination we should reuse the existent OpenSearch model with the elements 
`startIndex`, `totalResults`, `itemsPerPage`  

more info at [OpenSearch](https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md#the-totalresults-element)

**Comment by kbgg** [comment]
@m-mohr can you give an example of some environments where those values wouldn't be available? I included `returned` since that was already being returned in the sat-api implementation but I also don't see a use for it.

@p3dr0 from a quick glance it looks like OpenSearch is in XML format. What would be the benefit of following that standard? In STAC we're using underscore case for the key names and it looks like OpenSearch uses camel case so it would require us to diverge from our naming conventions just for the searching metadata.

**Comment by m-mohr** [comment]
@kbgg Any DMS that is only streaming the results without going through all of them. Cassandra for example. So you don't have a count.

@p3dr0 I don't see the benefit of that. We don't use startIndex but page and as we don't implement OpenSearch any time soon, it doesn't make sense to follow a very small subset of OpenSearch. Would only make sense if we implement OpenSearch so that we can re-use tools... So I agree with @kbgg here.

**Comment by p3dr0** [comment]
@kbgg  The important point is to support the concept of paging by adopting the same properties. I'm fine if we have `start_index`, `total_results`, `items_per_page` 
Just a note, an OpenSearch search service can return any given encoding (e.g. HTML, XML, JSON) and a OpenSearch EO Catalogue will be able to return STAC compliant document response 
For more information and use cases please check the [work done](http://ceos.org/ourwork/workinggroups/wgiss/access/opensearch/) by the CEOS WGISS on this issue 

@m-mohr again, the benefit is to support paging (not OpenSearch). 
I'd not make a decision on a very specific implementation experience (e.g. cassandra) but I'd understand that this might lead us to make it an optional returned value

**Comment by m-mohr** [comment]
Yes, that's what I was heading to, making it optional.

**Comment by m-mohr** [comment]
Merged in PR #482.
